packages/jest-runner/src/run_test.js

```diff
@@ -117,22 +121,36 @@ export default function runTest(
     })
     .then(
       result =>
-        Promise.resolve().then(() => {
-          environment.dispose();
-          if (globalConfig.logHeapUsage) {
-            if (global.gc) {
-              global.gc();
+        Promise.resolve()
```

Comment:
Could we instead use the `async` modifier on the promise callback and use `await`? I think that would be tons clearer for this code!